src/gui/Src/Gui/LogView.cpp

```diff
@@ -415,6 +420,27 @@ void LogView::copyToDebuggeeNotes()
     emit Bridge::getBridge()->setDebuggeeNotes(Notes);
 }
 
+void LogView::searchInLogSlot() {
+    dialogSearchInLog->show();
+
+    if (dialogSearchInLog->exec() == QDialog::Accepted) {
+        QList<QTextEdit::ExtraSelection> extraSelections;
+        QColor highlight = QColor(Qt::blue);
+
+        moveCursor(QTextCursor::Start);
+
+        // finding all occurances matching the regex given from start
+        while(find(QRegExp(dialogSearchInLog->editText))) {
+            QTextEdit::ExtraSelection extra;
+            extra.format.setBackground(highlight);
+            extra.cursor = textCursor();
+            extraSelections.append(extra);
+        }
+        // highlighting all those selections
+        setExtraSelections(extraSelections);

```

Comment:
It would also be good to at least scroll to the first occurrence (unless I missed it) and perhaps if you search the same again start searching from the selection (similar to the notepad find).